Server/src/services/DataUpload/EditUploadService.ts

```diff
@@ -0,0 +1,95 @@
+import { BadRequest, InternalServerError } from '@tsed/exceptions';
+import { getConnection } from "typeorm"
+import { IResponse } from "../../genericInterfaces/ResponsesInterface"
+import { DatasetCommonModel } from '../../models/DatasetModels/DatasetCommonModel';
+import { DatasetDeleteModel } from '../../models/DatasetModels/DatasetDeleteModel';
+import { DataUploadModel } from "../../models/DataUploadModel"
+import { Authors } from "../../models/entities/Authors"
+import AbstractUploadService from "./AbstractUploadService"
+
+
+export default class EditUploadService extends AbstractUploadService {
+    async uploadData(): Promise<IResponse> {
+        let deleteModel = new DatasetDeleteModel()
```

Comment:
is there a better way of updating a dataset than to delete it all and re-add it? seems like this is a lot of work for what it might do

Reply:
Old values not dont need to be kept if they are not linked to any other datasets

Reply:
Editing a data set will always be a lot of work as you have to edit over a half a dozen database tables. This is the least amount of work required. Going the full edit route is not possible given how some entities, ex. authors, are referenced by multiple tables. You cannot edit an author that is being referenced by a different publication, you have to make a new entry.